tests/phpunit/api/v3/ContactTest.php

```diff
@@ -5196,11 +5196,27 @@ protected function doMerge(bool $isReverse = FALSE) {
    * out when we are.
    */
   public function testEmojiInWhereClause(): void {
+    $schemaNeedsAlter = \CRM_Core_BAO_SchemaHandler::databaseSupportsUTF8MB4();
+    if ($schemaNeedsAlter) {
+      CRM_Core_DAO::executeQuery("
```

Comment:
@seamuslee001 so my conclusion is we can't do this alter - I ended up ripping it out because I hit so many weird situations in various parts of the suite. I think in the end it's better to just let this test actually test non-utf8 on those matrix machines that don't support utf8mb4 rather than try to mock it

Reply:
Also worth noting - this used to not cause problems because `\CRM_Core_BAO_SchemaHandler::databaseSupportsUTF8MB4()` used to always incorrectly return FALSE

Reply:
So this is the exact same alter that is happening in the v4 version of this test

Reply:
@seamuslee001 yeah - which I think we should remove to get rid of bizarre fails like https://github.com/civicrm/civicrm-core/pull/20984

Reply:
@eileenmcnaughton that is only failing because the Civi::statics cache hasn't been reset after the alter. This value https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/SchemaHandler.php#L922 with the current test setup will still return TRUE even tho table has now changed to not support emojis hence my reseting of the civi statics in the apiv4 test which is all that is needed to fix it

Reply:
The apiv4 suite should probably unset `Civi::statics` like the main suite does - https://github.com/civicrm/civicrm-core/blob/c3ea4bd79fd546aa98f709a53353adb3a5a64322/tests/phpunit/CiviTest/CiviUnitTestCase.php#L384 or we will keep getting weirdness